internal/controller/rabbitmq/rabbitmqvhost_controller.go

```diff
@@ -0,0 +1,271 @@
+		return ctrl.Result{}, nil
+	}
+
+	return r.reconcileNormal(ctx, instance, h)
+}
+
+func (r *RabbitMQVhostReconciler) reconcileNormal(ctx context.Context, instance *rabbitmqv1.RabbitMQVhost, h *helper.Helper) (ctrl.Result, error) {
+	// Get RabbitMQ cluster
+	rabbit := &rabbitmqclusterv2.RabbitmqCluster{}
+	err := r.Get(ctx, types.NamespacedName{Name: instance.Spec.RabbitmqClusterName, Namespace: instance.Namespace}, rabbit)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(rabbitmqv1.RabbitMQVhostReadyCondition, condition.ErrorReason, condition.SeverityWarning, rabbitmqv1.RabbitMQVhostReadyErrorMessage, err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	// Get admin credentials
+	rabbitSecret, _, err := oko_secret.GetSecret(ctx, h, rabbit.Status.DefaultUser.SecretReference.Name, instance.Namespace)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(rabbitmqv1.RabbitMQVhostReadyCondition, condition.ErrorReason, condition.SeverityWarning, rabbitmqv1.RabbitMQVhostReadyErrorMessage, err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	// Create API client
+	tlsEnabled := rabbit.Spec.TLS.SecretName != ""
+	protocol := "http"
+	managementPort := "15672"
+	if tlsEnabled {
+		protocol = "https"
+		managementPort = "15671"
+	}
+	baseURL := fmt.Sprintf("%s://%s:%s", protocol, string(rabbitSecret.Data["host"]), managementPort)
+	apiClient := rabbitmqapi.NewClient(baseURL, string(rabbitSecret.Data["username"]), string(rabbitSecret.Data["password"]), tlsEnabled)
+
+	// Create vhost
+	vhostName := instance.Spec.Name
+	if vhostName == "" {
+		vhostName = "/"
+	}
+
+	if vhostName != "/" {
+		err = apiClient.CreateOrUpdateVhost(vhostName)
+		if err != nil {
+			instance.Status.Conditions.Set(condition.FalseCondition(rabbitmqv1.RabbitMQVhostReadyCondition, condition.ErrorReason, condition.SeverityWarning, rabbitmqv1.RabbitMQVhostReadyErrorMessage, err.Error()))
+			return ctrl.Result{}, err
+		}
+	}
+
+	instance.Status.Conditions.MarkTrue(rabbitmqv1.RabbitMQVhostReadyCondition, rabbitmqv1.RabbitMQVhostReadyMessage)
+	instance.Status.Conditions.MarkTrue(condition.ReadyCondition, condition.ReadyMessage)
+
+	return ctrl.Result{}, nil
+}
+
+func (r *RabbitMQVhostReconciler) reconcileDelete(ctx context.Context, instance *rabbitmqv1.RabbitMQVhost, h *helper.Helper) (ctrl.Result, error) {
+	Log := log.FromContext(ctx)
+
+	// If TransportURL finalizer exists, wait for TransportURL to remove it
+	// The TransportURL controller manages this finalizer
+	if controllerutil.ContainsFinalizer(instance, rabbitmqv1.TransportURLFinalizer) {
+		return ctrl.Result{RequeueAfter: time.Duration(2) * time.Second}, nil
+	}
+
+	// Check for user-managed finalizers that may be blocking deletion
+	// These follow the pattern: rabbitmquser.rabbitmq.openstack.org/user-<name>
+	// We implement garbage collection to handle orphaned finalizers from force-deleted users
+	for _, finalizer := range instance.Finalizers {
```

Comment:
wondering if we need this since the user controller should remove the finalizer from the vhost when deleted? shouldn't the vhost just wait for that?